community/procedure-compiler/processor/src/main/java/org/neo4j/tooling/procedure/visitors/ContextFieldVisitor.java

```diff
@@ -69,36 +81,55 @@ private static String types( Set<Class<?>> supportedTypes )
     {
         if ( !hasValidModifiers( field ) )
         {
-            return Stream.of( new FieldError( field,
-                    "@%s usage error: field %s#%s should be public, non-static and non-final", Context.class.getName(),
-                    field.getEnclosingElement().getSimpleName(), field.getSimpleName() ) );
+            return Stream.of( new ContextFieldError( field,
+                    "@%s usage error: field %s should be public, non-static and non-final", Context.class.getName(),
+                    fieldFullName( field ) ) );
         }
 
         return Stream.empty();
     }
 
     private Stream<CompilationMessage> validateInjectedTypes( VariableElement field )
     {
-        if ( skipContextWarnings )
+        TypeMirror fieldType = field.asType();
+        if ( injectsAllowedType( fieldType ) )
         {
             return Stream.empty();
         }
 
-        TypeMirror fieldType = field.asType();
-        if ( !injectsAllowedTypes( fieldType ) )
+        if ( injectsRestrictedType( fieldType ) )
         {
-            return Stream
-                    .of( new ContextFieldWarning( field, "@%s usage warning: found type: <%s>, expected one of: %s",
-                            Context.class.getName(), fieldType.toString(), types( SUPPORTED_TYPES ) ) );
+            if ( ignoresWarnings )
+            {
+                return Stream.empty();
+            }
+
+            return Stream.of( new ContextFieldWarning( field, "@%s usage warning: found restricted type <%s> on %s.\n" +
```

Comment:
This should be changed to `@%s usage warning: found unsupported restricted type <%s> on %s.` instead, or something similar, since these APIs aren't official supported APIs and are subject to change at any time.